### DISCUSSION TITLE
pytest.ini log_date_format doesn't support %f directive

### DISCUSSION THREAD

**Discussion by ushankax** [pytest-dev/pytest#9551] [Ideas]
Hi, ladies and gentlemen!

Some people solve the problem of microseconds in pytest.ini with `%(msecs)03d` directive in `log_format` parameter. But there are cases when you need to specify microseconds in `log_date_format` and `%(msecs)03d` doesn't support there. For example, I work with logs data with **caplog** fixture. I get date from **LogRecord** object like `LogRecord.asctime`. It returns the format specified in `log_date_format`, so without `%f` it seems impossible to get microseconds.

Docs says that `log_date_format` is [strf-compatible](https://docs.python.org/3/library/time.html#time.strftime), no microseconds there too.

I look at it like at suggestion to include support of `%f` in `log_date_format`. 

P.S. Sorry if my English might seem bad, I tried my best 😄

**Comment by RonnyPfannschmidt** [comment]
based on https://docs.python.org/3/library/logging.html#logging.Formatter.formatTime it seems like this cant be supported nicely in a easy way

we may need a additional config options for the milisecond value

**Reply by ushankax** [reply]
As I understand it, it needs to edit `default_msec_format` or `default_time_format` depending on `%f` parameter contains or not.
